### PR TITLE
Transform

### PR DESCRIPTION
**Pull request by tomphp**
An unnecessary/complex dependency?

I use this extensively but maybe it adds confusing for people who are not used to it?

### CODE REVIEW

**Comment by localheinz** [comment]
Blocked by https://github.com/tomphp/config-service-provider/pull/37.

File: src/Pimple/PimpleContainerAdapter.php [hunk 1]
```diff
@@ -68,17 +68,22 @@ private function addServiceToContainer(ServiceDefinition $definition)
...
-            function ($argument) {
-                if (isset($this->container[$argument])) {
-                    return $this->container[$argument];
-                }
-
-                return $argument;
-            },
-            $arguments
-        );
+        return array_map([$this, 'resolveArgument'], $arguments);
```
Comment (localheinz) [inline src/Pimple/PimpleContainerAdapter.php]: @tomphp 

Ha, noticed this library when you tweeted about it some time ago. Not entirely sure, as it makes it a bit difficult renaming methods, but I'll leave it up to you.

**Comment by localheinz** [comment]
@tomphp 

:+1: Raised a concern, but feel free to merge!

File: src/Pimple/PimpleContainerAdapter.php (hunk 1 above)
Comment (tomphp) [inline src/Pimple/PimpleContainerAdapter.php]: @localheinz 

I use this library extensively in projects and I'm a big user of map/reduce/filter. That said, I'm not sure it's helpful here as it's only used twice. Let's leave it for now.

Comment (localheinz) [inline src/Pimple/PimpleContainerAdapter.php]: @tomphp 

👍

**Comment by tomphp** [comment]
Not adding this dependency for now.

**Pull request closed without merging**